### PR TITLE
feature: add variable type generation

### PR DESCRIPTION
**Pull request by nsalamad**
# Changes
* Add a `generate` command
    * Add a `types` command for generating types for our JS SDKs

### CODE REVIEW

File: docs/generate.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+`dvc generate`
+==============
+
+Generate Devcycle related files
+
+* [`dvc generate types`](#dvc-generate-types)
+
+## `dvc generate types`
+
+Generate Variable Types from the management API
```
Comment (ajwootto) [inline docs/generate.md:10]: maybe specify that this is typescript specifically?

**Review by suthar26 — APPROVED** [review]

File: src/api/variables.ts [hunk 2]
```diff
@@ -10,10 +10,20 @@ export class Variable {
...
 
 export const variableTypes = ['String', 'Boolean', 'Number', 'JSON']
 
+type ValidationSchema = {
+    schemaType: 'enum' | 'regex' | 'jsonSchema'
+    enumValues?: string[] | number[]
+    regexPattern?: string
+    jsonSchema?: string
+    description: string
+    defaultValue: string | number | { [key: string]: string | boolean | number }
```
Comment (suthar26) [inline src/api/variables.ts:24]: do we need boolean here? also number is duplicated

File: src/commands/generate/types.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          if (err instanceof Error) message = err.message
+          this.writer.failureMessage(`Unable to write to ${flags['output-dir']}/dvcVariableTypes.ts` + `: ${message}`)
+      }
+  }
+
+  private getTypesString(variables: Variable[], react: boolean): string {
+      let types =
+          'type DVCJSON = { [key: string]: string | boolean | number }\n\n' +
+          `${!react ? 'export ' : ''}type DVCVariableTypes = {\n` +
+          `${variables.map((variable) => `    ${this.getVariableKeyAndtype(variable)}`).join('\n')}` +
```
Comment (suthar26) [inline src/commands/generate/types.ts:64]: can we change the spaces to `\t`

File: test/commands/generate-types.test.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+const artifactsDir = './test/artifacts/'
+const jsOutputDir = artifactsDir + 'generate/js'
+const reactOutputDir = artifactsDir + 'generate/react'
+
+const expectedTypesString =
+`type DVCJSON = { [key: string]: string | boolean | number }
+
+export type DVCVariableTypes = {
+    'enum-var': 'Hello' | 'Hey' | 'Hi'
```
Comment (suthar26) [inline test/commands/generate-types.test.ts:59]: could change these spaces to also be `\t` if tests fail

**Pull request merged**